client-rpc/src/rpc/sync_rpc.rs

```diff
@@ -73,31 +73,60 @@ pub trait SyncRpc: Send + Sync {
     fn sync_stop(&self, request: WalletRequest) -> Result<()>;
 }
 
-pub struct SyncRpcImpl<S, C, O>
+pub struct SyncRpcImpl<S, C, O, T>
 where
     S: Storage,
     C: Client,
     O: TransactionObfuscation,
+    T: WalletClient,
```

Comment:
`WalletClient` is a big trait, normally `WalletClient` calls the syncer, rather than reverse.

Reply:
it's for calling wall service, i'll find a way not to to use WalletClient

Reply:
that's because of WalletClient
fn recover_addresses
it need current addresses, and also generate,
it need to use WalletClient